app/src/main/java/com/infomaniak/mail/views/itemViews/BannerWithActionView.kt

```diff
@@ -0,0 +1,97 @@
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.view.isVisible
+import com.infomaniak.lib.core.utils.getAttributes
+import com.infomaniak.lib.core.utils.setMarginsRelative
+import com.infomaniak.mail.R
+import com.infomaniak.mail.databinding.ViewBannerWithActionBinding
+
+class BannerWithActionView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0,
+) : ConstraintLayout(context, attrs, defStyleAttr) {
+
+    private val binding by lazy { ViewBannerWithActionBinding.inflate(LayoutInflater.from(context), this, true) }
+
+    var description: CharSequence?
+        get() = binding.description.text
+        set(value) {
+            binding.description.text = value
+        }
+
+    var actionButtonText: CharSequence?
+        get() = binding.actionButton.text
+        set(value) {
+            binding.actionButton.text = value
+        }
+
+    private val baseConstraints by lazy {
+        ConstraintSet().apply {
+            clone(binding.root)
+        }
+    }
+    private val buttonAlignConstraints by lazy {
+        ConstraintSet().apply {
+            clone(baseConstraints)
+            connect(R.id.description, ConstraintSet.END, R.id.actionButton, ConstraintSet.START)
+            connect(R.id.actionButton, ConstraintSet.START, R.id.description, ConstraintSet.END)
+            connect(R.id.actionButton, ConstraintSet.END, ConstraintSet.PARENT_ID, ConstraintSet.END)
+            connect(R.id.actionButton, ConstraintSet.TOP, ConstraintSet.PARENT_ID, ConstraintSet.TOP)
+            connect(R.id.actionButton, ConstraintSet.BOTTOM, ConstraintSet.PARENT_ID, ConstraintSet.BOTTOM)
+        }
+    }
+
+    init {
+        attrs?.getAttributes(context, R.styleable.BannerWithActionView) { initView(attributes = this) }
+    }
+
+    fun setOnActionClickListener(callback: () -> Unit) {
+        binding.actionButton.setOnClickListener { callback() }
+    }
+
+    fun getGoneHandle() = binding.goneHandle
+
+    private fun initView(attributes: TypedArray) {
+        with(binding) {
+            if (attributes.getBoolean(R.styleable.BannerWithActionView_isButtonAlignWithDescription, false)) {
+                binding.description.setMarginsRelative(end = resources.getDimensionPixelSize(R.dimen.alternativeMargin))
+                buttonAlignConstraints.applyTo(root)
+            }
+
+            attributes.getDrawable(R.styleable.BannerWithActionView_descriptionIcon)?.let {
+                descriptionIcon.isVisible = true
+                descriptionIcon.setCompoundDrawablesWithIntrinsicBounds(it, null, null, null)
+            }
+
+            actionButton.icon = attributes.getDrawable(R.styleable.BannerWithActionView_buttonIcon)
+        }
+
+        description = attributes.getString(R.styleable.BannerWithActionView_description) ?: ""
+        actionButtonText = attributes.getString(R.styleable.BannerWithActionView_buttonText) ?: ""
+    }
```

Comment:
If you want to move the init of the view in its own method rather than doing it directly inside `init`. Place it right after `init` so we can easily find it at the top of the class like other custom views